typing/primitive.ml

```diff
@@ -24,7 +24,7 @@ type native_repr =
   | Same_as_ocaml_repr
   | Unboxed_float
   | Unboxed_integer of boxed_integer
-  | Untagged_int
+  | Untagged_int of Path.t
```

Comment:
I haven't looked at this part of the codebase recently, and it is not clear to me why you want to have this `Path.t` constructor here. Can you explain? It looks like you are trying to make it possible to go back to the original type from the `native_repr` information, but (1) you don't actually achieve this here, as you forget some type parameters for non-constant constructors, and (2) I don't see why that would be required. Naively I would expect that just keeping `Untagged_immediate` and considering that all untagged-immediate types have the same (equal) 'native representation' should work.

Reply:
I  get you point with parametric type, so I should fix that. 
The problem was the function [Primitive.equal_native_repr], which I did not know really where it is used. 
Seems it is just checking that the annotation are the same between signature and structure. As the type are unified anyway, we don't need what I am doing. You are right.

I was also thinking abour renaming Untagged_int to Untagged_immediate. I fix all this now.

Reply:
I applied your suggestions... the integration test are running.